homeassistant/components/switch_as_x/light.py

```diff
@@ -100,3 +110,7 @@ def async_state_changed_listener(event: Event | None = None) -> None:
 
         # Call once on adding
         async_state_changed_listener()
+
+        # Add this entity to the wrapped switch's device
+        registry = er.async_get(self.hass)
+        registry.async_update_entity(self.entity_id, device_id=self._device_id)
```

Comment:
Yes, we can do that.

Then we should also:
a) Remove the `switch_as_x` from the device if the other config entry is removed from the device
b) Remove the `switch_as_x` from the device if the tracked entity is removed from the device

Is some other additional cleanup is needed?
- If the other config entry is removed, it will be removed from the device, so that's covered by a)
- When our config entry is removed, it will be removed from the device automatically by `DeviceRegistry.async_clear_config_entry`

Reply:
Sounds good.

Reply:
OK, implemented.
It added quite a lot of code though.